Libraries/PeriphDrivers/Include/MAX32657/i3c.h

```diff
@@ -0,0 +1,621 @@
+ *
+ */
+struct _i3c_ccc_req_t {
+    uint8_t ccc; ///< CCC command to send.
+    uint8_t target_addr : 7; ///< Target address if CCC is a direct command. Ignored
+    ///< if CCC is a broadcast command.
+    uint8_t read_write : 1; ///< Set to 1 for read commands and set to 0 for write
+    ///< commands. Ignored for broadcast CCCs.
+    uint8_t flags; ///< See request flags above.
+    uint8_t def_byte; ///< Optional defining byte. Defined by CCC.
+    uint8_t subCmd; ///< Optional sub-command. Defined by CCC. Only used by direct
+    ///< CCCs.
+    unsigned char *tx_buf; ///< Optional data bytes to send.
+    uint8_t tx_len; ///< Length of optional data.
+    unsigned char *rx_buf; ///< Optional data bytes to read.
+    uint8_t rx_len; ///< Length of optional data.
+};
+
+/**
+ * @brief Private SDR request structure.
+ *
+ * SDR read and write to an I3C target.
+ *
+ */
+struct _i3c_req_t {
+    uint8_t target_addr : 7; ///< Target address.
+    uint8_t is_i2c : 1; ///< If this is a legacy I2C transfer.
+    uint8_t stop; ///< Send a STOP after the transaction.
+    unsigned char *tx_buf; ///< Optional data bytes to send.
+    uint16_t tx_len; ///< Length of optional data.
+    unsigned char *rx_buf; ///< Optional data bytes to read.
+    uint16_t rx_len; ///< Length of optional data.
+};
+
+/**
+ * @brief   IBI payload request callback. Write additional byte to \a byte.
+ *
+ * This function will be called as long as non-zero is returned.
+ *
+ * @return  Non-zero if a byte is written to \a byte, 0 to indicate no more additional
+ * bytes left to send.
+ */
+typedef int (*mxc_i3c_ibi_getbyte_t)(mxc_i3c_regs_t *i3c, unsigned char *byte);
+
+/***** Function Prototypes *****/
+
+/* ************************************************************************* */
+/* Control/Configuration functions                                           */
+/* ************************************************************************* */
+
+/**
+ * @brief   Initialize and enable I3C peripheral.
+ *
+ * @note    On default this function enables I3C peripheral clock and I3C gpio pins.
+ *          If you wish to skip clock and gpio initialization, define MSDK_NO_GPIO_CLK_INIT
+ *          flag in project.mk file.
+ *
+ * @param   i3c         Pointer to I3C registers (selects the I3C block used).
+ * @param   targetMode  Whether to put the device in controller or target mode. Use
+ *                      non-zero.
+ * @param   ppHz        I3C push-pull frequency. Only valid in controller mode.
+ * @param   odHz        I3C open-drain frequency. Only valid in controller mode.
+ * @param   i2cHz       I2C frequency. Only valid in controller mode.
+ * @param   staticAddr  I2C-style static address. Has no effect if the address is hardwired.
+ *
+ * @return  Success/Fail, see \ref MXC_Error_Codes for a list of return codes.
+ */
+int MXC_I3C_Init(mxc_i3c_regs_t *i3c, int targetMode, uint8_t staticAddr, uint32_t ppHz,
```

Comment:
Please wrap the initialization args in a struct.  This will help avoid some maintenance issues in the future if we decide to change or add options.

i.e.

```C
typedef struct {
    bool targetMode;
    uint8_t staticAddr;
    // ...
} mxc_i3c_config_t;

int MXC_I3C_Init(mxc_i3c_regs_t *i3c, mxc_i3c_config_t *config_options);
```